toolchain/parse/node_kind.def

```diff
@@ -681,6 +681,35 @@ CARBON_PARSE_NODE_KIND_BRACKET(InterfaceDecl, InterfaceIntroducer,
                                CARBON_TOKEN(Semi)
                                    CARBON_IF_ERROR(CARBON_TOKEN(Interface)))
 
+// `impl ... as`:
+//     ImplIntroducer
+//     _repeated_ _external_: AccessModifierKeyword or DeclModifierKeyword
+//         _external_: ImplicitParamList
+//       ImplForall
+//     _optional_
```

Comment:
I was thinking about this, and I wonder if it'd be clearer to do something like:

(below):

```
// `forall ...`:
//   _external_: ImplicitParamList
// ImplForall
```

and here:

```
//     _optional_: ImplForAll (see below)
```

Reply:
Done.